task/task.js

```diff
@@ -1,9 +1,9 @@
 export class Task {
-    constructor(text = "", done = false, onUpdate, onDelete, onDone, onUnDone) {
+    constructor(text = "", done = false, eventHandlers) {
+        Object.assign(this, eventHandlers);
```

Comment:
Extract the specific events you expect to receive from eventHandlers, and assign them only.